test/TestArrayUtils.cpp

```diff
@@ -0,0 +1,1214 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+#define GPU_ACTIVE
+
+// std library headers
+#include <array>
+
+// other library headers
+#include "gtest/gtest.h"
+
+// care headers
+#include "care/array.h"
+#include "care/array_utils.h"
+#include "care/care.h"
+
+// Array Fill Tests
+TEST(array_utils, fill_empty)
+{
+   care::host_ptr<int> a;
```

Comment:
Which ArrayFill method is being called with a host_ptr? Does it call the local_ptr method? I'm thinking we should make these host_device_ptrs.